server/routes/dsl_route.ts

```diff
@@ -46,13 +46,19 @@ export function registerDslRoute(router: IRouter) {
                 size,
                 body: rest,
               };
-
         const start = performance.now();
         try {
-          const resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
-            'search',
-            params
-          );
+          let resp;
+          if(dataSourceEnabled && dataSourceId1){
+            const client = context.dataSource.opensearch.legacy.getClient(dataSourceId1);
+            resp = await client.callAPI('search', params);
+          }
+          else{
+              resp = await context.core.opensearch.legacy.client.callAsCurrentUser(
```

Comment:
trying to understand this logic, while if enable datasource and there is dataSourceId1, wait for the search in dataSourceId1, but is it when there is a datasource, then the enableDataSource is true. 

this is dataSourceEnable=!!datasource [here](https://github.com/opensearch-project/dashboards-search-relevance/blob/aafcda0644f78575473a9f2e194688f3ae5d98a9/public/application.tsx#L26)

is the check for datasourceEnabled still needed?

Reply:
if  its enabled and datasourceid is not present it needs to follow the old logic for registering the client, thats why the check